pyccel/ast/numpyext.py

```diff
@@ -142,21 +142,41 @@ class NumpyReal(PythonReal):
     1.0
     """
 
-PrecisionToDtype_int = {
-    4 : NumpyInt32,
-    8 : PythonInt,
+#==============================================================================
+PrecisionToDtype = {
+    'int' : {
+        4 : NumpyInt32,
+        8 : PythonInt},
+    'float' : {
+        4 : NumpyFloat32,
+        8 : PythonFloat},
+    'complex' : {
+        4 : NumpyComplex64,
+        8 : PythonComplex,
+        16 : NumpyComplex128,}
 }
 
-PrecisionToDtype_float = {
-    4 : NumpyFloat32,
-    8 : PythonFloat,
-}
+def process_stype(stype, precision):
+    """
+    returns a cast function from
+    PrecisionToDtype_int, PrecisionToDtype_float, PrecisionToDtype_complex, python_builtin_datatypes
+    by ginving it:
+    stype:
+        supported data type name.
+    precision:
+        describs the precision of stype, to make the returned function more specific.
```

Comment:
This docstrings needs better formatting, as usual:
```python
"""
[Function description]

Parameters
----------
stype : str
     [parameter description]

precision : int
     [parameter description]

Results
-------
<base class of all return objects> or None
    [result description]
"""
```
See https://numpydoc.readthedocs.io/en/latest/format.html#sections.

Further, there are some typos to be fixed:
- ginving -> giving
- describs -> describes